tests/ui/impl-trait/precise-capturing/rpitit-outlives.rs

```diff
@@ -0,0 +1,15 @@
+//@ check-pass
+
+#![feature(precise_capturing_in_traits)]
+
+struct Invariant<T>(*mut T);
+
+trait Foo {
+    fn hello<'s: 's>(&'s self) -> Invariant<impl Sized + use<Self>>;
+}
+
+fn hello<'s, T: Foo + 'static>(x: &'s T) -> Invariant<impl Sized + 'static> {
```

Comment:
Wrapping in invariant here is necessary to actually make these outlives errors lol. Otherwise bivariance is just "too chill" and we don't actually end up constraining `'s: 'static`.

Reply:
please add a comment explaining this test, iirc it checks that bivariant lifetimes are ignored when computing type outlives components?